mycpp/gc_alloc.h

```diff
@@ -147,11 +166,21 @@ inline Str* NewStr(int len) {
 // s->MaybeShrink() afterward!
 inline Str* OverAllocatedStr(int len) {
   int obj_len = kStrHeaderSize + len + 1;  // NUL terminator
-  void* place = gHeap.Allocate(sizeof(ObjHeader) + obj_len);
+  const size_t num_bytes = sizeof(ObjHeader) + obj_len;
+#if MARK_SWEEP
+  int obj_id;
+  bool in_pool;
+  void* place = gHeap.Allocate(num_bytes, &obj_id, &in_pool);
```

Comment:
I wonder if there is some way for Allocate() to fill in the header directly, and then `T::obj_header()` only fills in the changed ones ...

Can save that for later ...